diagnostics/block.go

```diff
@@ -0,0 +1,150 @@
+package diagnostics
+
+import (
+	"container/list"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/ledgerwatch/erigon-lib/metrics"
+)
+
+type MilliSeconds time.Duration
+
+func (m MilliSeconds) MarshalJSON() ([]byte, error) {
+	return json.Marshal(time.Duration(m).Milliseconds())
+}
+
+var (
```

Comment:
consider the following naming improvements:
```
headerDelayGauge = metrics.NewGauge(`block_production_delays{type="header"}`)
bodyDelayGauge = metrics.NewGauge(`block_production_delays{type="body"}`)
preExecutionDelayGauge = metrics.NewGauge(`block_production_delays{type="pre_execution"}`)
executionDelayGauge = metrics.NewGauge(`block_production_delays{type="execution"}`)
blockProductionExecutionDelayGauge = metrics.NewGauge(`block_production_delays{type="block_production_execution"}`)
```

remarks:
1. would be good to avoid the use of "average" in the metric name since this is not an average value that we are recording but an observation
2. the name of the metric is more descriptive - block_production_delays and so are the types, e.g. type=block_production_execution